modules/core/arrow-typeclasses/src/main/kotlin/arrow/typeclasses/Hash.kt

```diff
@@ -0,0 +1,45 @@
+package arrow.typeclasses
+
+/**
+ * A type class used to represent hashing for objects of type [F]
+ *
+ * @see // Fancy link to docs
+ */
+interface Hash<in F> : Eq<F> {
+
+  /**
+   * Produces a hash for an object of type [F].
+   *
+   * @receiver The object to hash
+   * @returns an int representing the object hash
+   */
+  fun F.hash(): Int
```

Comment:
Use kdoc embedded ank snippets to show each one of the public api methods with examples. In the next version of Arrow these will show up in the browser as runnable:
```
/**
   * Commented method or class
   * 
   * {: data-executable='true'}
   *
   * ```kotlin:ank
   * import arrow.aql.instances.list.select.*
   * import arrow.aql.instances.listk.select.select
   *
   * fun main(args: Array<String>) {
   *   //sampleStart
   *   val result: List<Int> =
   *     listOf(1, 2, 3).query {
   *       select { this + 1 }
   *     }.value()
   *  //sampleEnd
   *  println(result)
   *  }
   *  ```
   */
```

Reply:
I ment actually in the sources too. The example will then show up in the API docs beside the docs page.

Reply:
Yeah, that confused me a bit :) 
Added it now for `F.hash()`, `Hash.invoke()` and `Hash.any()`